File: python/2020/july/add-digits.py
class Solution:
    def add_digits(self, num):
        if num == 0:
            return 0
        else:
            return (num - 1) % 9 + 1


def add_digits(num):
    while num >= 10:
        tmp = 0
        while num > 0:
            tmp += num%10
            num //=10
        num = tmp
    return num

File: code/java/src/main/java/cn/az/code/year2020/july/AddDigits.java
package cn.az.code.year2020.july;

/**
 * @author az
 * @since 07/26/20
 */
public class AddDigits {

    public static void main(String[] args) {
        AddDigits ad = new AddDigits();
        System.out.println(ad.addDigits(1267));
    }

    public int addDigits(int num) {
        String numStr = String.valueOf(num);
        if (numStr.length() == 1) {
            return Integer.parseInt(numStr);
        }
        int temp = 0;
        for (char c : numStr.toCharArray()) {
            temp += (c - 48);
        }
        return addDigits(temp);
    }

}
